hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/subscription/SubscriptionService.java

```diff
@@ -94,17 +110,49 @@ public SubscriptionService(SubscriptionRepository subscriptionRepository,
 
     public void createSubscription(Subscription subscription, String createdBy, CreatorRights creatorRights) {
         subscriptionValidator.checkCreation(subscription, creatorRights);
-        subscriptionRepository.createSubscription(subscription);
+        multiDcExecutor.execute(new CreateSubscriptionRepositoryCommand(subscription));
         auditor.objectCreated(createdBy, subscription);
         subscriptionOwnerCache.onCreatedSubscription(subscription);
     }
 
     public Subscription getSubscriptionDetails(TopicName topicName, String subscriptionName) {
-        return subscriptionRepository.getSubscriptionDetails(topicName, subscriptionName).anonymize();
+        Subscription subscription = subscriptionRepository.getSubscriptionDetails(topicName, subscriptionName)
+                .anonymize();
+        subscription.setState(getEffectiveState(topicName, subscriptionName));
+        return subscription;
+    }
+
+    private Subscription.State getEffectiveState(TopicName topicName, String subscriptionName) {
+        Set<Subscription.State> states = loadSubscriptionStatesFromAllDc(topicName, subscriptionName);
```

Comment:
how about logging a warning if the set size is > 1 (which means something is out of sync for this sub)?

Reply:
`logger.warn("Some states are out of sync: {}", states);` ?

Reply:
Yep, something alike. But with full subscription name included and operation related.